common_types/src/datum.rs

```diff
@@ -1085,4 +1243,101 @@ mod tests {
             assert!(source.to_negative().is_none());
         }
     }
+
+    #[test]
+    fn test_parse_datum_date_from_str() {
+        let cases = ["-9999-01-01", "9999-12-21", "0-1-1", "1000-02-28"];
+
+        let expects = [
+            Datum::Date(-4371587),
+            Datum::Date(2932886),
+            Datum::Date(-719528),
+            Datum::Date(-354227),
+        ];
+
+        for (i, source) in cases.iter().enumerate() {
+            let datum = Datum::parse_datum_date_from_str(source).unwrap();
+            assert_eq!(datum, expects.get(i).unwrap().clone());
+        }
+
+        let err_cases = [
+            "ab-01-01",
+            "01-ab-01",
+            "-9999-234-ab",
+            "100099-123-01",
+            "1990-01-123",
+        ];
+
+        for source in err_cases {
+            Result::unwrap_err(Datum::parse_datum_date_from_str(source));
+        }
```

Comment:
As for the error cases, we can test it in another test.

Reply:
done